### PR TITLE
DietPi-Software | Samba: Move disk cache to RAM

### PR DESCRIPTION
**Pull request by MichaIng**
**Status**: Ready
- [x] Changelog

**Testing**:
- [x] Install
- [x] Service
- [x] Reboot
- [x] Reinstall
- [x] Uninstall

🈯️ Passed on Jessie, Stretch and Buster

**Reference**: https://github.com/Fourdee/DietPi/issues/2396

### CODE REVIEW

File: dietpi/dietpi-software [hunk 1]
```diff
@@ -3344,6 +3344,17 @@ _EOF_
...
+
+			# Link disk cache to RAM: https://github.com/Fourdee/DietPi/issues/2396
+			# - Remove previous disk cache dir or symlink
+			[[ -e /var/cache/samba ]] && rm -R /var/cache/samba
+			# - Pre-create RAM cache dir
+			mkdir -p /var/run/samba-cache
+			# - Link disk cache to RAM
+			ln -s /var/run/samba-cache /var/cache/samba
+			# - Create RAM cache dir automatically on boot
+			echo 'd /var/run/samba-cache - - - - -' > /etc/tmpfiles.d/dietpi-samba_cache.conf
```
Comment (Fourdee) [inline dietpi/dietpi-software:3356]: @MichaIng 

Nice 👍 

Reading this, does it actually make a tmpfs file, as ```/var``` is mounted to disk?
https://developers.redhat.com/blog/2016/09/20/managing-temporary-files-with-systemd-tmpfiles-on-rhel7/

If not, we might be better off using ```/tmp```?

Comment (WolfganP) [reply dietpi/dietpi-software:3356]: /var/run is linked to /run which is a tmpfs already.

Comment (Fourdee) [reply dietpi/dietpi-software:3356]: @WolfganP 

Legend 👍 💘 
```
root@Fuzon:~# ls -lha /var/run
lrwxrwxrwx 1 root root 4 Jul 21 21:56 /var/run -> /run
```

**Comment by Fourdee** [comment]
@MichaIng @WolfganP 

Looks good from my end, nice implementation. @WolfganP ?

**Comment by WolfganP** [comment]
> Looks good from my end, nice implementation. @WolfganP ?

@Fourdee @MichaIng LGTM, couldn't test on my main system as is not migrated yet, but I tested manually and worked OK

File: dietpi/dietpi-software (hunk 1 above)
Comment (MichaIng) [reply dietpi/dietpi-software:3356]: @Fourdee 
Jep, `tmpfiles` has nothing to do with `tmpfs` (RAMdisk). But as stated, /var/run -> /run is tmpfs as well.

**Pull request merged**